ykcompile/src/lib.rs

```diff
@@ -480,50 +494,35 @@ impl<TT> TraceCompiler<TT> {
         true
     }
 
+    /// Emit code for a runtime memcpy(3).
+    fn c_memcpy(&mut self, dest: &RegAndOffset, src: &RegAndOffset, size: u64) {
+        let sym = Self::find_symbol("memcpy").unwrap();
+        self.caller_save();
+        dynasm!(self.asm
+            ; push rax
+            ; xor rax, rax
+            ; lea rdi, [Rq(dest.reg) + dest.offs]
+            ; lea rsi, [Rq(src.reg) + src.offs]
+            ; mov rdx, size as i32
+            ; mov r11, QWORD sym as i64
+            ; call r11
```

Comment:
Why is it necessary to lookup the `memcpy` fn pointer, put it in a register and then call it. Can't you just call memcpy directly here?

Reply:
I've been unable to achieve that using `dynasm` so far. Looking at the docs, it seems you can't.

![image](https://user-images.githubusercontent.com/604955/94848384-75ab9180-0413-11eb-946b-44ead6f77e38.png)


Notice there is no `imm64` variant. [This SO post](
https://stackoverflow.com/questions/19552158/call-an-absolute-pointer-in-x86-machine-code) confirms it too.

I think you can use `rel32` [if you know where memcpy is relative to the instruction pointer](https://stackoverflow.com/questions/38961192/how-to-execute-a-call-instruction-with-a-64-bit-absolute-address)

Reply:
Huh, strange. That stack overflow post was interesting.